### PR TITLE
feature(topics): add topic tags

### PR DESCRIPTION
**Pull request by arathjz**
**What:**
Add tags on topics

**Why:**
To address: https://app.asana.com/0/1159164196409129/1186091925043867/f

**How:**
- Add category on a topic title
- Add tags on a topic title

**Media**
image
image

### CODE REVIEW

File: javascripts/discourse/widgets/dc-topic-title.js.es6 [hunk 1]
```diff
@@ -5,7 +5,43 @@ import { h } from "virtual-dom";
 export default createWidget("dc-topic-title", {
   tagName: "div.dc-topic-title",
 
+  buildCategory(attrs) {
```
Comment (duranmla) [inline javascripts/discourse/widgets/dc-topic-title.js.es6:8]: What are the issues you encountered for not to use a built-in widget to render the category?
_I just wanted to make sure you evaluate the possibility of adding a tag using Discourse core but you couldn't_

File: scss/widgets/post.scss [hunk 2]
```diff
@@ -120,6 +120,42 @@
...
+        width: 0.5rem;
+        height: 0.5rem;
+        margin-right: 0.35rem;
+      }
+    }
+
+    .dc-topic-tag {
+      font-size: 0.8rem;
+      border-radius: 0.25rem;
+      border: 1px solid $gray-400;
```
Comment (duranmla) [inline scss/widgets/post.scss:148]: Keep using `rem` even for the border definition. `0.0625rem`

File: scss/widgets/post.scss [hunk 3]
```diff
@@ -120,6 +120,42 @@
     margin: 0 0 ($spacer * 0.5) 0;
     line-height: 1;
   }
+
+  .dc-topic-info {
+    margin-bottom: 0.5rem;
+
+    .dc-topic-category,
+    .dc-topic-tag {
+      color: $gray-800 !important;
```
Comment (duranmla) [inline scss/widgets/post.scss:129]: As this is a sort of new widget the `!important` is really need it?

File: scss/widgets/post.scss [hunk 4]
```diff
@@ -120,6 +120,42 @@
...
+      margin-right: 0.5rem;
+      display: inline-block;
+    }
+
+    .dc-topic-category {
+      font-size: 0.9rem;
```
Comment (duranmla) [inline scss/widgets/post.scss:135]: Within `variables.scss` file we have:

```
$font-size-base
$font-size-lg
$font-size-sm
$font-size-xs
```

Keep using standard sizes whenever is possible. I see here `0.9rem` and then `0.8rem` maybe we can keep like: `$font-size-sm` and `$font-size-xs` respectively?

_Also the bottom line here is not about creating more variables for different sizes but to use a defined set of sizes to keep UI consistent_

File: javascripts/discourse/widgets/dc-topic-title.js.es6 [hunk 5]
```diff
@@ -5,7 +5,43 @@ import { h } from "virtual-dom";
...
+      h('span', {
+        style: {
+          background: `#${categoryColor}`
+        }
+      }),
+      categoryName
+    ]);
+  },
+
+  buildTopicTags(attrs) {
```
Comment (duranmla) [inline javascripts/discourse/widgets/dc-topic-title.js.es6:27]: Probably very related to pass comment for category:
> _What are the issues you encountered for not to use a built-in widget to render the tags?_

File: scss/widgets/post.scss (hunk 3 above)
Comment (arathjz) [reply scss/widgets/post.scss:129]: Yeah, actually it is not needed

File: javascripts/discourse/widgets/dc-topic-title.js.es6 (hunk 1 above)
Comment (arathjz) [reply javascripts/discourse/widgets/dc-topic-title.js.es6:8]: Yes, actually that's the case. This was the only way I found to perform this task, we may do some pair programming to considerate some other approach. What do you think?

Comment (duranmla) [reply javascripts/discourse/widgets/dc-topic-title.js.es6:8]: We can check it out afterwards as a side activity to understand a bit more the `whys` of this. _(I think I have an idea)_ but in any case for now I think we are good to go!

**Pull request merged**